liblwgeom/cunit/cu_algorithm.c

```diff
@@ -1229,6 +1230,19 @@ static void test_median_robustness(void)
 	/* Unsupported geometry type */
 	do_median_test("POLYGON((1 0,0 1,1 2,2 1,1 0))", NULL, LW_TRUE, 1000);
 	do_median_test("POLYGON((1 0,0 1,1 2,2 1,1 0))", NULL, LW_FALSE, 1000);
+
+	/* Intermediate point included in the set */
+	do_median_test("MULTIPOINT ZM ("
+			"(0 0 20000 0.5),"
+			"(0 0 59000 0.5),"
+			"(0 48000 -20000 1.3),"
+			"(0 -48000 -20000 1.3),"
+			"(0 -3000 -3472.22222222222262644208967685699462890625 1),"
+			"(0 3000 3472.22222222222262644208967685699462890625 1),"
+			"(0 0 -1644.736842105263121993630193173885345458984375 1),"
```

Comment:
I love this one. Really great, @Algunenano :)

Reply:
Sad news. For some reason (I'm guessing different float operation order) this hits in clang (5.0) but not in gcc (7.2.1, 6.4.1). I'm investigating the issue to try to find a common case

Reply:
Good news :smile: The issue was due to invalid rounding (-0.00000000000029917588874109483636135671651489707215 vs 0.0) in `init_guess` with the  `-ffloat-store` flag  that I guess clang just ignores.
I'll have a look to see if the flag is really useful.